pimd/pim_sock.c

```diff
@@ -313,12 +420,90 @@ int pim_socket_join(int fd, struct in_addr group, struct in_addr ifaddr,
 	}
 
 	return ret;
+
+}
+
+static inline int pim_socket_join_v6(int fd, pim_addr group, pim_addr ifaddr,
+			      ifindex_t ifindex)
+{
+	int ret;
+
+	struct ipv6_mreq opt;
```

Comment:
I think, instead of introducing two new functions for V4 & V6, we can utilize these macros introduced in @eqvinox 's PR (https://github.com/FRRouting/frr/pull/10356) to keep the code same
* PIM_ADDRSTRLEN ==> for INET_ADDRSTRLEN/INET6_ADDRSTRLEN
* PIM_AF ==> for AF_INET/AF_INET6

Also the opt definition, we can try to do this way

```
#if PIM_IPV == 4

#ifdef HAVE_STRUCT_IP_MREQN_IMR_IFINDEX
	struct ip_mreqn opt;
#else
	struct ip_mreq opt;
#endif
	opt.imr_multiaddr = group;
#ifdef HAVE_STRUCT_IP_MREQN_IMR_IFINDEX
	opt.imr_address = ifaddr;
	opt.imr_ifindex = ifindex;
#else
	opt.imr_interface = ifaddr;
#endif

#else /* V6 */

        struct ipv6_mreq opt;
	memcpy(&opt.ipv6mr_multiaddr,&group,sizeof(struct in6_addr));
	opt.ipv6mr_interface = ifindex;
#endif
```

Reply:
The reason for having 2 functions is because lot of #ifdefs are already used in these functions and i felt it affects readability of code with more #ifdefs. Hence created a separate function for v6 and made one function which gets exposed outside to other files.